docs/reference/ingest/ingest-node.asciidoc

```diff
@@ -721,12 +721,29 @@ All processors are defined in the following way within a pipeline definition:
 // NOTCONSOLE
 
 Each processor defines its own configuration parameters, but all processors have
-the ability to declare `tag` and `on_failure` fields. These fields are optional.
+the ability to declare `tag` ,`on_failure` and `if` fields. These fields are optional.
 
 A `tag` is simply a string identifier of the specific instantiation of a certain
 processor in a pipeline. The `tag` field does not affect the processor's behavior,
 but is very useful for bookkeeping and tracing errors to specific processors.
 
+The `if` field must contain a script that returns a boolean value. If the script evaluates to `true`
+then the processor will be executed for the given document otherwise it will be skipped.
+The `if` field takes a map with the script settings used defined in <<script-processor, script-options>>
```

Comment:
`if` doesn't take a map, right? I think you mean "object" (since we are talking about json), and that is optional (the example you have below here has `if` as the script source directly. You would only need an object if you wanted to pass params (unlikely for ingest I think) or use a different scripting language (not really possible since expressions don't support this, but theoretically a native script could be written as a custom script engine).

Reply:
Right .. "object" wins :)

Hmm on that note and kinda off topic but: maybe we should/could enable expressions here the same way we do for bucket selector aggregation (interpret `1.0` as `true`) to become consistent there?

Reply:
I think it would be difficult and not very useful, since expressions only have access to numerics. Expressions operate on doc values, but ingest does not have doc values, so it would require a lot of hacking.

Reply:
ok makes sense :)